### PR TITLE
Apply DSS on prognostic fields using callback operation 

### LINKED ISSUE #270: Add dss to state callback

### PR DESCRIPTION
**Pull request by akshaysridhar**
Adds a new callback method for DSS on state variables. Closes #270 

modified:   src/Callbacks/Callbacks.jl
modified:   src/Callbacks/callback.jl
modified:   src/Simulations/Simulations.jl
modified:   src/Simulations/simulation.jl

- [ ] Integration tests

### CODE REVIEW

**Comment by akshaysridhar** [comment]
I'd like to run longer simulations (All box LES for instance) before merging - but please check the general structure + default application of this callback in the meantime

File: src/Callbacks/callback.jl [hunk 1]
```diff
@@ -1,3 +1,34 @@
...
+prognostic variables at user defined timestep interval. 
+Recommended timestep interval is 1.
+"""
+struct StateDSS{M <: AbstractModel, I <: Number} <: AbstractCallback
+    model::M
+    interval::I
+end
+function (F::StateDSS)(integrator)
+    Y = integrator.u
+    for i in propertynames(Y)
```
Comment (akshaysridhar) [inline src/Callbacks/callback.jl:14]: @jiahe23 Is there a cleaner way of accessing all the state fields from the simulation object ?

Comment (jiahe23) [reply src/Callbacks/callback.jl:14]: I don't know any better ways...

**Review by szy21 — APPROVED** [review]

File: src/Simulations/Simulations.jl [hunk 2]
```diff
@@ -2,12 +2,13 @@ module Simulations
...
 import OrdinaryDiffEq
 const ODE = OrdinaryDiffEq
+using OrdinaryDiffEq: CallbackSet
 
 using JLD2
 using Printf
 using UnPack
 using ClimaCore: Fields
-using ..Models, ..Callbacks, ..Domains
+using ..Models, ..Callbacks, ..Domains, ..Callbacks
```
Comment (akshaysridhar) [inline src/Simulations/Simulations.jl]: Remove duplicate

**Comment by bischtob** [comment]
@akshaysridhar , status?

**Comment by jiahe23** [comment]
closes #270

**Pull request closed without merging**